src/scripts/Docs/Type.elm

```diff
@@ -19,156 +25,44 @@ type Type
-            in
-            maybeAddParens (argsHtml ++ toHtml Func result)
-
-        Var name ->
-            [ text name ]
-
-        Apply name [] ->
-            [ text name.name ]
-
-        Apply name args ->
-            let
-                maybeAddParens =
-                    case context of
-                        Func ->
-                            identity
-
-                        App ->
-                            Code.addParens
-
-                        Other ->
-                            identity
-
-                argsHtml =
-                    List.concatMap (\arg -> space :: toHtml App arg) args
-            in
-            maybeAddParens (text name.name :: argsHtml)
-
-        Tuple args ->
-            List.map (toHtml Other) args
-                |> List.intersperse [ text ", " ]
-                |> List.concat
-                |> Code.addParens
-
-        Record fields ext ->
-            let
-                fieldsHtml =
-                    List.map fieldToHtml fields
-                        |> List.intersperse [ text ", " ]
-                        |> List.concat
-
-                recordInsides =
-                    case ext of
-                        Nothing ->
-                            fieldsHtml
-
-                        Just extName ->
-                            text extName :: text " | " :: fieldsHtml
-            in
-            text "{ " :: recordInsides ++ [ text " }" ]
-
-
-fieldToHtml : ( String, Type ) -> List (Html msg)
-fieldToHtml ( field, tipe ) =
-    text field :: space :: colon :: space :: toHtml Other tipe
-
-
-length : Context -> Type -> Int
-length context tipe =
-    case tipe of
-        Function args result ->
-            let
-                parens =
-                    case context of
-                        Func ->
-                            2
-
-                        App ->
-                            2
-
-                        Other ->
-                            0
-
-                argLengths =
-                    List.map (\t -> 4 + length Func t) args
-            in
-            parens + List.sum argLengths + length Func result
-
-        Var name ->
-            String.length name
-
-        Apply { name } [] ->
-            String.length name
-
-        Apply { name } args ->
-            let
-                parens =
-                    case context of
-                        Func ->
-                            0
-
-                        App ->
-                            2
+parse : String -> Result String Type
+parse =
+    Decode.decodeValue decoder << Encode.string
 
-                        Other ->
-                            0
 
-                argsLength =
-                    List.sum (List.map (\t -> 1 + length App t) args)
-            in
-            parens + String.length name + argsLength
+decoder : Decoder Type
+decoder =
+    let
+        toInternal functionArgs elmType =
```

Comment:
Should this go into the top level?
I don't expect that code to live too long though. The `distance` function should use the new "official" types soonish, so I don't have strong feelings.